src/coreclr/src/vm/readytoruninfo.h

```diff
@@ -18,17 +18,43 @@
 
 typedef DPTR(struct READYTORUN_SECTION) PTR_READYTORUN_SECTION;
 
+class NativeImage;
 class PrepareCodeConfig;
 
+typedef DPTR(class ReadyToRunCoreInfo) PTR_ReadyToRunCoreInfo;
+class ReadyToRunCoreInfo
+{
+private:
+    PTR_PEImageLayout               m_pLayout;
+    PTR_READYTORUN_CORE_HEADER      m_pCoreHeader;
+    
+public:
+    ReadyToRunCoreInfo();
+    ReadyToRunCoreInfo(PEImageLayout * pLayout, READYTORUN_CORE_HEADER * pCoreHeader);
+
+    PEImageLayout * GetLayout() const { return m_pLayout; }
+    IMAGE_DATA_DIRECTORY * FindSection(ReadyToRunSectionType type) const;
+
+    PTR_PEImageLayout GetImage() const
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_pLayout;
+    }
+};
+
 typedef DPTR(class ReadyToRunInfo) PTR_ReadyToRunInfo;
 class ReadyToRunInfo
 {
     friend class ReadyToRunJitManager;
 
     PTR_Module                      m_pModule;
-
-    PTR_PEImageLayout               m_pLayout;
     PTR_READYTORUN_HEADER           m_pHeader;
+    bool                            m_isComponentAssembly;
+    NativeImage*                    m_pNativeImage;
```

Comment:
E.g. These need to be `PTR_NativeImage`, etc. like all other pointers in this structure to work correctly under DAC.